### ISSUE TITLE
feat: ML feature engineering pipeline for 1-min futures data

### ISSUE THREAD

**Issue by SantiBarrios2002** [SantiBarrios2002/hmm-momentum-trading#79]
## Context

To compare classical signal processing (HMM/IOHMM) against modern ML, we need a shared feature engineering pipeline. The features should include the same information available to the HMM models (returns, volatility, time-of-day) plus additional features that ML can exploit.

### `src/ml/features.py`

**Feature groups:**

1. **Return features** — lagged returns at multiple horizons
   - `Δy_{t-1}, Δy_{t-2}, ..., Δy_{t-L}` (L configurable, e.g. 10)
   - Cumulative returns over windows (5m, 15m, 30m, 60m)

2. **Volatility features** — same as IOHMM side info
   - EWMA vol ratio (σ_short / σ_long) — reuse from `side_info.py`
   - Realized vol over rolling windows
   - Return magnitude `|Δy_t|`

3. **Seasonality features**
   - Time-of-day (minutes since open, normalized)
   - Sin/cos encoding of time for cyclical representation

4. **Microstructure features**
   - Bid-ask spread proxy (high-low range)
   - Volume (if available in Databento data)

5. **Temporal split utilities**
   - `expanding_window_split(T, train_frac)` — for walk-forward validation
   - No data leakage: features computed only from past data

## Functions

```python
def build_feature_matrix(returns, timestamps, lag_window=10, vol_windows=(5,15,30,60)):
    """Build (T, D) feature matrix from 1-min returns and timestamps.
    
    All features are causal — computed from t-1 and earlier only.
    Returns X (T, D) and feature_names list.
    """

def temporal_train_test_split(X, y, train_frac=0.7):
    """Split respecting temporal order. No shuffling."""

def normalize_features(X_train, X_test):
    """Z-score normalize using train stats only (no leakage)."""
```

## Tests (`tests/test_ml_features.py`)

- [ ] Feature matrix shape: (T, D) with D = expected feature count
- [ ] No NaN in features after warm-up period
- [ ] Features are causal: feature at t depends only on data up to t-1
- [ ] Normalization uses only training stats
- [ ] Temporal split preserves order (max train timestamp < min test timestamp)

## Acceptance criteria

- [ ] All functions have docstrings
- [ ] All tests pass
- [ ] Works on Databento 1-min ES data

## Dependencies

- #75 (side info features — reuse EWMA vol ratio)
